eclipse-scout-core/src/util/ObjectUuidProvider.ts

```diff
@@ -0,0 +1,326 @@
+
+  /**
+   * Prefix for all UI generated IDs.
+   */
+  static UI_SEQ_ID_PREFIX = '_ui_'; // must not contain any dots ('.') so that the id can be used as css selector "#..." and for the RegExp 'UI_SEQ_PATTERN'.
+
+  /**
+   * Delimiter for the segments of a uuidPath.
+   */
+  static UUID_PATH_DELIMITER = '|'; // "-" is used by UUID, "." by ClassNames, "_" by ClassId path from Java (see ITypeWithClassId.ID_CONCAT_SYMBOL).
+
+  /**
+   * Delimiter for dependent uuids
+   */
+  static DEPENDENT_UUID_DELIMITER = '@';
+
+  /** use {@link createUiSeqId} to generate a new ID */
+  protected static _UI_SEQ_ID_PATTERN = new RegExp('^' + ObjectUuidProvider.UI_SEQ_ID_PREFIX + '\\d+$');
+  protected static _INSTANCE: ObjectUuidProvider;
+
+  /**
+   * Modifiable set of widgets which will be skipped when building the uuidPath.
+   * A widget is skipped if its class is exactly one of these (NOT instanceof!).
+   *
+   * A widget may be skipped if it is not relevant for computing the uuidPath, e.g. if it is only a layouting component.
+   * For example: A group box is skipped because the id or uuid of a widget is normally unique inside a form so the group box would unnecessarily enlarge the uuidPath.
+   * Also, if the widget is moved into another group box, the uuidPath won't be affected.
+   * If the group box is extracted into a separate widget and gets its own class (aka. template field)
+   * it must not be skipped anymore because this template can be used multiple times on the same form and must therefore be part of the uuidPath.
+   * This template use case is the reason why the subclasses of the registered widgets are not considered.
+   */
+  static uuidPathSkipWidgets: Set<Constructor<Widget>> = new Set<Constructor<Widget>>();
+
+  /**
+   * Modifiable list of rules which are used to determine if a parent should be skipped when building the uuidPath.
+   *
+   * These rules are always applied, even if {@link UuidPathOptions.considerSkipWidgets} is set to false.
+   */
+  static uuidPathAlwaysSkipRules: ((widget: Widget) => boolean)[] = [];
+
+  constructor() {
+    this.objectType = null;
+    this.id = null;
+  }
+
+  /**
+   * Computes a path starting with the {@link uuid} of this object. If a parent is available, its {@link uuidPath} is appended to the right (recursively).
+   * {@link UUID_PATH_DELIMITER} is used as delimiter between the segments.
+   * By default, if the object is a remote (Scout Classic) object having a classId, its value is directly returned without appending the parent path because classIds typically already include its parents.
+   *
+   * @param object The object for which the uuidPath should be computed.
+   * @param options Optional {@link UuidPathOptions} controlling the computation of the path.
+   *
+   * @returns the uuid path starting with this object's uuid or null if no path can be created.
+   */
+  uuidPath(object: ObjectUuidSource, options?: UuidPathOptions) {
+    options = scout.nvl(options, {});
```

Comment:
Maybe specify the default value in the method signature instead.

Reply:
scout.nvl also considers null in addition to undefined. So uuidPath(object, null) would cause an error. This is why I don't use it very often. We should discuss it